BB-2659 juniper xblock html and xblock completable

taniwha opened the pull request:
This PR updates xblock-html for juniper.

**JIRA tickets**: https://tasks.opencraft.com/browse/BB-2659

**Dependencies**: None

**Screenshots**: N/A

**Sandbox URL**: N/A

**Merge deadline**: None

**Testing instructions**:

1. Set up a juniper devstack
2. Run `make requirements` and `make test`

**Author notes and concerns**:

1. Surely there's more to it, especially testing.

**Reviewers**
- [ ] @lgp171188 
- [ ] edX reviewer[s] TBD

lgp171188 commented on requirements/test.txt:
```diff
@@ -1,14 +1,14 @@
 # Github requirements
-git+https://github.com/edx/django-pyfs.git@1.0.3#egg=django-pyfs==1.0.3
+git+https://github.com/edx/django-pyfs.git@2.2#egg=django-pyfs==2.2
 git+https://github.com/edx/xblock-sdk.git#egg=xblock-sdk
```
@taniwha, `xblock-sdk` can now be installed from PyPi. See [this](https://github.com/edx/xblock-sdk/releases/tag/0.2.0).

lgp171188 commented on requirements/base.txt:
```diff
@@ -1,7 +1,7 @@
-edx-i18n-tools==0.4.7
-Mako==1.0.7
-bleach==2.1.4
+git+https://github.com/edx/xblock-utils.git@2.0.0#egg=xblock-utils==2.0.0
+django-statici18n==1.9.0
+edx-i18n-tools==0.5.3
+Mako==1.1.3
+bleach==3.1.5
```
@taniwha, can we try to use the same versions of the libraries used by the Juniper release?

```suggestion
git+https://github.com/edx/xblock-utils.git@2.1.1#egg=xblock-utils==2.1.1
django-statici18n==1.9.0
edx-i18n-tools==0.5.3
Mako==1.1.3
bleach==3.1.5
```

lgp171188 commented on requirements/quality.txt:
```diff
@@ -1,9 +1,9 @@
+pydocstyle==5.0.2
+pylint==2.4.4
+caniusepython3==7.2.0
+edx_lint==1.5.0
```
```suggestion
edx-lint==1.4.1
```

lgp171188 commented on requirements/test.txt:
```diff
@@ -1,14 +1,14 @@
-lazy==1.1
-pytest-cov==2.6.0
-tox==3.1.3
-tox-battery==0.5.1
+lazy==1.4
+pytest-cov==2.10.0
+tox==3.16.1
+tox-battery==0.6.1
 mock==3.0.5
 
 # Github requirements
-git+https://github.com/edx/django-pyfs.git@1.0.3#egg=django-pyfs==1.0.3
+git+https://github.com/edx/django-pyfs.git@2.2#egg=django-pyfs==2.2
```
@taniwha, again, while it is not a strict requirement, it might be better to use similar depedency versions as Juniper.

lgp171188 commented on requirements/base.txt:
```diff
@@ -1,7 +1,7 @@
 # Requirements for app run
```
@taniwha, you will also have to pin Django using `django~=2.2` because one of the dependencies that requires Django only specifies the minimum Django version.

lgp171188 commented on requirements/quality.txt:
```diff
@@ -1,9 +1,9 @@
-caniusepython3==7.0.0
-edx_lint==0.5.5
+isort==4.3.21
+astroid==2.3.3
```
@taniwha, you can update the CircleCI configuration to use the Python 3.5 image instead of Python 2.7.

taniwha commented:
@lgp171188 I believe I have addressed all of your points. If the version bump is not sufficient, let me know and I'll adjust both it and the reference in the readme.

lgp171188 commented on README.md:
```diff
@@ -18,6 +18,8 @@ pip install https://github.com/open-craft/xblock-html
 You may specify the `-e` flag if you intend to develop on the repo.
 
+Note that as of version 0.2.0, Python 2.7 is no longer supported. The current minimum Python version is 3.5.
+
```
@taniwha, the `README` file doesn't give the full installation/usage instructions - for example, it doesn't specify what needs to be added to the Studio advanced modules list to be able to use the HTML5 and the excluded HTML5 XBlocks. Can you add the details for those?

lgp171188 commented on html_xblock/html.py:
```diff
@@ -59,7 +59,7 @@ def resource_string(path):
+        Return a fragment that contains the html for the student view.
         """
         frag = Fragment()
         frag.content = xblock_loader.render_template('static/html/lms.html', {'self': self})
```
@taniwha, the below warning message is printed when running the tests. Can we fix this and the 2 other occurrences?

```
UserWarning: ResourceLoader.render_template has been deprecated in favor of ResourceLoader.render_django_template
```

Agrendalath commented on setup.py:
```diff
@@ -23,7 +23,7 @@ def package_data(pkg, roots):
 setup(
     name='html-xblock',
-    version='0.1.3',
+    version='0.2.0',
```
This is a breaking change, so we should bump MAJOR. [Ref](https://semver.org/).

lgp171188 replied:
@Agrendalath, I mentioned this in one of my review comments. But I wasn't sure if we could call this XBlock 1.0, as that version has some implicit expectations about the quality, maturity, stability etc.

@taniwha, can you bump the major version in `setup.py`?
